lib/active_model/serializer/reflection.rb

```diff
@@ -42,13 +42,13 @@ class Serializer
       def build_association(subject, parent_serializer_options)
         association_value = subject.send(name)
         reflection_options = options.dup
-        serializer_class = ActiveModel::Serializer.serializer_for(association_value, reflection_options)
+        serializer_class = subject.class.serializer_for(association_value, reflection_options)
```

Comment:
`serializer_context_class`/`parent_serializer` (however we may call it) is not a reflection option but a serializer option. I agree that there is room for improvement though.